src/test/java/tests/TurboIssueEventTests.java

```diff
@@ -0,0 +1,400 @@
+                createLabelUpdateEvent("C", IssueEventType.Labeled,
+                        new GregorianCalendar(2015, 1, 1, 2, 30, 15),
+                        "C1"));
+        sampleEvents.add(
+                createLabelUpdateEvent("D", IssueEventType.Unlabeled,
+                        new GregorianCalendar(2015, 1, 1, 3, 20, 59),
+                        "D1"));
+    }
+
+    /**
+     * The method groupLabelUpdateEvents should not modify the input events list
+     */
+    @Test
+    public void testLabelUpdateEventsGroupingInvariant() {
+        List<TurboIssueEvent> events = new ArrayList<>(sampleEvents);
+        TurboIssueEvent.groupLabelUpdateEvents(events);
+        assertEquals(sampleEvents, events);
+    }
+
+    /**
+     * Test the events grouping when list of events is empty
+     */
+    @Test
+    public void testLabelUpdateEventsGroupingForNoEvent() {
+        assertEquals(new ArrayList<TurboIssueEvent>(),
+                     TurboIssueEvent.groupLabelUpdateEvents(new ArrayList<TurboIssueEvent>()));
+    }
+
+    /**
+     * Test the events grouping when there is only 1 event
+     */
+    @Test
+    public void testLabelUpdateEventsGroupingForOneEvent() {
+        List<TurboIssueEvent> events = new ArrayList<>();
+        events.add(sampleEvents.get(0));
+
+        List<TurboIssueEvent> expectedSubList1 = new ArrayList<>();
+        expectedSubList1.add(sampleEvents.get(0));
+
+        List<List<TurboIssueEvent>> expected = new ArrayList<>();
+        expected.add(expectedSubList1);
+
+        assertEquals(expected,
+                     TurboIssueEvent.groupLabelUpdateEvents(events));
+    }
+
+    /**
+     * Test when A and B update the overlapping set of labels
+     * at overlapping time. The order of modification are
+     * A:+L1 -> B:-L1 -> A:+L2 -> B:+L2
+     * Expected: [A:[+L1, +L2], B:[-L1, +L2]]
+     * Note: This scenario may not actually occur since A and B
+     * might not be recorded as adding the same label L2 by github
+     */
+    @Test
+    public void testLabelUpdateEventsGrouping1() {
+        List<TurboIssueEvent> events = new ArrayList<>();
+
+        events.add(createLabelUpdateEvent("A", IssueEventType.Labeled,
+                                          new GregorianCalendar(2015, 1, 1, 1, 1, 0),
+                                          "L1"));
+        events.add(createLabelUpdateEvent("B", IssueEventType.Unlabeled,
+                                          new GregorianCalendar(2015, 1, 1, 1, 1, 25),
+                                          "L1"));
+        events.add(createLabelUpdateEvent("A", IssueEventType.Labeled,
+                                          new GregorianCalendar(2015, 1, 1, 1, 1, 40),
+                                          "L2"));
+        events.add(createLabelUpdateEvent("B", IssueEventType.Labeled,
+                                          new GregorianCalendar(2015, 1, 1, 1, 2, 11),
+                                          "L2"));
```

Comment:
I didn't check. Do you incorporate boundary cases? i.e. events happening at 59,60,61 seconds apart.